modules/programs/neomutt-accounts.nix

```diff
@@ -30,7 +30,7 @@ with lib;
     };
   };
 
-  config = {
+  config = mkIf config.notmuch.enable {
```

Comment:
Shouldn't this be `config.neomutt.enable`?

Reply:
no; its like for alot, neomutt needs notmuch to work so instead of adding yet another enable option; I just used the noitmuch one.

Reply:
But neomutt works just fine without notmuch (that's how I used it). So I can imagine that people might be surprised that they actually need to enable notmuch for the account to be available in neomutt. There is nothing in the module descriptions to indicate that this is something one needs to do.

My suggestion is to either

- add a `neomutt.enable` account option (and activate neomutts notmuch support if also `notmuch.enable` is set) or

- describe somewhere, perhaps in the `programs.neomutt.enable` option that the module currently assumes that accounts where notmuch is enables also should be available in neomutt.

I'm somewhat in favor of the first option since somebody might want to use neomutt without notmuch and to accomodate them we'd have to add the `neomutt.enable` account option and to do that after the fact without breaking existing configurations we'd have to introduce a state version change.

Even if the module only works when notmuch is enabled I'd suggest adding the  `neomutt.enable` account option and add an assertion raising an error if notmuch is not enabled. Then somebody who'd like to use neomutt without notmuch could identify the issue and contribute a module improvement to also allow non-notmuch usage.

Reply:
I see, my belief was that neomutt started as a way to apply patches refused by mutt, notably the notmuch ones. Makes sense you could use it as a standalone. I've added an option as you state. The notmuch string is only added if notmuch is enabled so no need for an extra warning. The test passes.